### PR TITLE
WT-7837 Clear updates structure in wt_hs_insert_updates after processing each key

### PR DESCRIPTION
**Pull request by mm-ng**

### CODE REVIEW

**Review by quchenhao — APPROVED** [review]
lgtm

**Comment by kommiharibabu** [comment]
@mm-ng Can you write a python test to demonstrate this problem? WT lack of tests in this area leads to a BF.

File: test/suite/test_hs25.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, nrows + 1):
+            self.session.begin_transaction()
+            cursor.set_key(str(i))
+            cursor.set_value(str(i) + value)
+            self.assertEqual(cursor.update(), 0)
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(commit_ts))
+        cursor.close()
+
+    def prepare_updates(self, uri, value, nrows, prepare_ts):
```
Comment (quchenhao) [inline test/suite/test_hs25.py:21]: I won't do a prepared update on each key. Instead I would do a prepared update every two updates.

Comment (quchenhao) [reply test/suite/test_hs25.py:21]: The problem only triggers with a non prepared key follows a prepared key.

Comment (quchenhao) [reply test/suite/test_hs25.py:21]: Also no need to use a lot of keys. I think two keys is enough to reproduce this.

File: test/suite/test_hs25.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import wttest
```
Comment (kommiharibabu) [inline test/suite/test_hs25.py:1]: The test is missing the file header. There is also another test test_hs24.py also missed a file header. Can you fix that also as part of this ticket?

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**